core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/samModels.scala

```diff
@@ -175,6 +186,31 @@ object AppAction {
     sealerate.collect[AppAction].map(a => (a.asString, a)).toMap
 }
 
+sealed trait WorkspaceAction extends Product with Serializable {
+  def asString: String
+}
+object WorkspaceAction {
+  final case object CreateControlledResource extends WorkspaceAction {
+    val asString = "create_controlled_application_private"
+  }
+
+  val allActions = sealerate.values[WorkspaceAction]
+  val stringToAction: Map[String, WorkspaceAction] =
+    sealerate.collect[WorkspaceAction].map(a => (a.asString, a)).toMap
+}
+
+sealed trait WsmResourceAction extends Product with Serializable {
+  def asString: String
+}
+object WsmResourceAction {
+  final case object Write extends WsmResourceAction {
+    val asString = "write"
```

Comment:
So when we create resources, we give the `creator` the wsm `editor` permission, which automatically gives them `read` and `write` roles in sam. So, if we assume leo is creating them, users should have both. I'll add read here and use it as the permission checked in `getRuntime` though since it seems more semantically correct